askama/src/lib.rs

```diff
@@ -76,23 +76,50 @@ pub use askama_escape::{Html, Text};
 pub trait Template {
     /// Helper method which allocates a new `String` and renders into it
     fn render(&self) -> Result<String> {
-        let mut buf = String::with_capacity(self.size_hint());
+        let mut buf = String::with_capacity(Self::SIZE_HINT);
         self.render_into(&mut buf)?;
         Ok(buf)
     }
     /// Renders the template to the given `writer` buffer
-    fn render_into(&self, writer: &mut dyn std::fmt::Write) -> Result<()>;
+    fn render_into<W: std::fmt::Write + ?Sized>(&self, writer: &mut W) -> Result<()>;
+
+    /// The template's extension, if provided
+    const EXTENSION: Option<&'static str>;
+
+    /// Provides an conservative estimate of the expanded length of the rendered template
+    const SIZE_HINT: usize;
+}
+
+pub trait DynTemplate {
+    fn dyn_render(&self) -> Result<String>;
```

Comment:
Maybe copy the Template's doc strings verbatim for the dyn_render() and dyn_render_into()?

Reply:
Yup.